plugin/ranger.vim

```diff
@@ -23,36 +23,42 @@
 
 
 " ================ Ranger =======================
+if !exists('g:ranger_path')
+    let g:ranger_path = 'ranger'
+endif
+
 if has('nvim')
-  function! OpenRanger()
-    let currentPath = expand("%:p:h")
-    let rangerCallback = { 'name': 'ranger' }
-    function! rangerCallback.on_exit(id, code)
-      Bclose!
-      try
-        if filereadable('/tmp/chosenfile')
-          exec system('sed -ie "s/ /\\\ /g" /tmp/chosenfile')
-          exec 'argadd ' . system('cat /tmp/chosenfile | tr "\\n" " "')
-          exec 'edit ' . system('head -n1 /tmp/chosenfile')
-          call system('rm /tmp/chosenfile')
-        endif
-      endtry
+    function! OpenRanger(dir)
+        let currentPath = expand(a:dir)
+        let tmp_file_path = tempname()
+        let rangerCallback = { 'name': 'ranger' , 'tmp_file_path': tmp_file_path}
+        function! rangerCallback.on_exit(id, code)
+            bdelete!
```

Comment:
I use neovim and do not have `Bclose`, and @bramblex's changes work perfectly.  In fact the `tabnew` change makes it function in a much more useful way.

It's hard to explain what it's even doing, because it isn't actually opening things in new tabs.  It's opening them where I expect them.  If I make a new split, and then open ranger from that buffer, the selected file from ranger is opened inside that buffer.